src/graphene-rect.c

```diff
@@ -756,6 +778,41 @@ graphene_rect_round (const graphene_rect_t *r,
   res->size.height = ceilf (res->size.height);
 }
 
+/**
+ * graphene_rect_round_extents:
```

Comment:
I'm not entirely sure what's the purpose of this function, and how it relates to `graphene_rect_round()`. Care to explain a bit more? Maybe write an example in the documentation, so that it's clear which function should be used, and when.

Reply:
So... As I wrote briefly in the PR description and hopefully a bit more in the commits, when you've cases in which you want to be sure that the rounded rectangle covers completely the area that was covered by the fractional rectangle (i.e. you want to damage that are) you need to make sure that you recompute the sizes taking in account the ceiled bottom-right angle coordinates.

Otherwise if the difference between the float coords and the floored coords is major than the difference between the ceiled size and and the fractional size, then we'd have a left/top shift which is not compensated by the size increase.

This happens also in the the round test, where the rectangle is properly rounded, and in fact its area is bigger than the original one, but the rounded one won't contain the original, because the left shift isn't covered enough by the size ceiling.

My initial thought was to just fix the `round` method, but since the APi was clearly done assuming only that we're flooring and ceilling, then I decided to add a different method for this.

So, said this, I'd be happy to be helped in wording this properly for the docs :)

Reply:
⇑ What you wrote here should go in the description of the function. :smiley: 

> If you want to be sure that the rounded rectangle completely covers
> the area that was covered by the original rectangle—i.e. you want to
> cover the area including all its corners—this function will make sure
> that the size is recomputed taking into account the ceiling of the
> coordinates of the bottom-right corner. If the difference between the
> original coordinates and the coordinates of the rounded rectangle is
> greater than the difference between the original size and and the
> rounded size, then the move of the origin would not be compensated
> by a move in the anti-origin, leaving the corners of the original rectangle
> outside the rounded one.

Does that sound good?

I'd also honestly be happy with fixing the `round()` method, to be quite honest. Yes, it says what it does in the documentation, but it also says that the new rectangle would cover the old one, and that's clearly not always the case.

I'd also be happy with deprecating `round()` after adding `round_extents()`, if we want to play it extra-safe.

Reply:
As per the choice of deprecating `round()` or fixing it, I'm fine with the 2nd one.

But since you're the maintainer, and I don't want to risk introducing something that could break assumptions taken by other toolkits, I added a new one.

So let me know what you prefer. I can add the deprecation commit for now. If you then want to instead fix, I can take the other direction.

Reply:
This looks good, thanks!